addons/mail/models/mail_channel.py

```diff
@@ -498,9 +501,15 @@ def _channel_message_notifications(self, message, message_format=False):
             :returns list of bus notifications (tuple (bus_channe, message_content))
         """
         message_format = message_format or message.message_format()[0]
+        channel_last_message_ids = dict((r['id'], r['message_id']) for r in self._channel_last_message_ids())
         notifications = []
         for channel in self:
-            notifications.append([(self._cr.dbname, 'mail.channel', channel.id), dict(message_format)])
+            notification_content = dict(message_format)
+            notification_content['channel_info'] = {
+                "last_message_id": channel_last_message_ids.get(channel.id, False),
+                "message_unread_counter": channel.message_unread_counter
```

Comment:
I don't think it's right adding this here?
Channel notif will be sent to all users, but the unread_counter is user specific.

Reply:
Didn't saw it was build for all users so yes indeed it smells... If we can't return the unread counter here then I'm wondering if the whole refactoring with the python server being more cooperative makes sense. Indeed the fact that it returns this value at some points and not all is a problem and adding a covered point does not really help... Do you have an idea @seb-odoo ?

Reply:
I think in the context of this PR we should probably keep the local counter in JS and send info from the server only when it is possible / makes sense.

I still have trouble to understand why it is problematic that it doesn't return the value always. We build everything with the principle of incomplete information.

Reply:
The problem is that you can have two different situations that, given the info you have, are totally identical. Sometimes you need to trust server value, sometimes not.
I think we should go back to what was done before this PR and just change JS so it does not flicker or we need to make server return needed info at **each point** to the right user(s) but any mixed situations is problematic for me.

If you want I can take the time to write a scenario to show the problems raised by mixed situations but it'll take time as I need to search back what I've done before the last commits. Maybe it could become a "big" test but I'm not sure it's easily translatable to a test.